examples/mundialis_extended.py

```diff
@@ -0,0 +1,146 @@
+    temporal_extent -- Temporal Extent -- 44/DateRange -- 2018-05-01T00:00:00.000Z/2018-10-01T00:00:00.000Z
+    
+    Outputs:
+    band_dir -- Band Directory -- 45/User String
+    band_files -- Band Files -- 45/User String
+    out_dir -- directory to store output
+    
+    Main Dependency:
+    python-3
+
+    Software Dependencies:
+    openeo-0.4
+    
+    Processing Resources:
+    ram -- 1
+    disk -- 10
+    cpu -- 1
+    """
+    all_bands = all_bands.split()
+    bands = bands.split()
+    temporal_extent = temporal_extent.split("/")
+    west, south, east, north = list(map(lambda value: value.strip(), bbox_string.split(",")))
+    bbox = {"west": west, "east": east, "south": south, "north": north, "crs": "EPSG:4326"}
+    logger.info("Demo user:password: %s:%s", user, password)
+    logger.info("Provider: %s with driver URL %s", provider, driver_url)
+    logger.info("Image Collection: %s", image_collection)
+    logger.info("Bands: %s", bands)
+    logger.info("Using BBox in str format - raw input (west, south, east, north): %s", bbox_string)
+    logger.info("BBox: %s", bbox)
+    logger.info("Temporal extent in string format start/end: %s", temporal_extent)
+    connection = openeo.connect(driver_url)
+    connection.authenticate_basic(user, password)
+
+    logger.info(
+        "describe_collection('%s'):\n %s\n",
+        image_collection,
+        json.dumps(connection.describe_collection(image_collection), indent=2),
+    )
+    cube = ImageCollectionClient.load_collection(session = connection, collection_id = image_collection, bands = all_bands)
+    cube = cube.filter_bbox( **bbox)
+    cube = cube.filter_temporal(extent=temporal_extent)
+
+    logger.info("cube.graph: \n%s\n", json.dumps(cube.graph, indent=2))
+
+    logger.info("File Format: %s", band_format)
+    logger.info(
+        "File Name Format: {provider}.{image_collection}.{west}.{south}.{east}.{north}."
+        "{temporal_extent[0]}.{temporal_extent[1]}.B{band.zfill(2)}.{band_format}"
+    )
+    band_dir = out_dir
+    logger.info("Downloading bands in %s", out_dir)
+    band_files = []
+    for band in bands:
+        cube_band = cube.band(band)
+        band_file = (
+            f"{provider}.{image_collection}.{west}.{south}.{east}.{north}."
+            f"{temporal_extent[0]}.{temporal_extent[1]}.B{band.zfill(2)}.{band_format.lower()}"
+        )
+        logger.info("Downloading band %s: %s", band, band_file)
+        band_path = f"{out_dir}/{band_file}"
+        band_files.append(band_file)
+        logger.info("Starting download at %s", datetime.now())
+        cube_band.download(band_path, format=band_format)
+        logger.info("Download finished for band %s at %s", band, datetime.now())
+
+    logger.info("Downloads finished at %s", datetime.now())
+
+    return {"band_dir": band_dir, "band_files": band_files}
+
+
+def example():
+    execute(
+        all_bands="S2_2 S2_3 S2_4 S2_5 S2_6 S2_7 S2_8A S2_11 S2_12",
```

Comment:
why not specifying the bands as a python list like `["S2_2", "S2_3", "S2_4", "S2_5", "S2_6"]` ? It seems cumbersome to pass around strings and postpone the responsibility to parse them properly

Reply:
nevermind, I see the goal is to receive this data directly from command line